src/main/java/org/mars/rover/kata/MarsNavigator.java

```diff
@@ -1,60 +1,72 @@
 package org.mars.rover.kata;
 
-import org.mars.rover.kata.commands.CommandParser;
 import org.mars.rover.kata.entrydata.CommandSet;
 import org.mars.rover.kata.entrydata.RoverInstructions;
 import org.mars.rover.kata.entrydata.StdinProcessor;
+import org.mars.rover.kata.location.Coordinate;
+import org.mars.rover.kata.location.Grid;
+import org.mars.rover.kata.location.OccupiedArea;
+import org.mars.rover.kata.location.Position;
 
-import java.io.ByteArrayInputStream;
 import java.util.ArrayList;
 import java.util.List;
-import java.util.Scanner;
+import java.util.stream.IntStream;
 
 public class MarsNavigator {
     CommandSet providedInput;
 
-    ArrayList<ArrayList<Coordinate>> grid;
+    Grid grid;
 
     List<MarsRover> marsRovers;
 
     public MarsNavigator(CommandSet providedInput) {
         this.providedInput = providedInput;
-        this.grid = new ArrayList<>();
+        this.grid = Grid.createGrid(providedInput.gridX(), providedInput.gridY());
```

Comment:
Czytelniej IMHO by wyglądało `Grid.createGrid(providedInput.width(), providedInput.height())`. To punkt czy koordynata przyjmuje X i Y.